user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.js

```diff
@@ -412,27 +440,35 @@ define([
       }
 
       fileButtons.canDownload(this.get("canDownload"));
-      //TODO handle file button press
+
       var filePath = clickedFile.obj.attr("path");
-      filePath = Encoder.encodeRepositoryPath(filePath);
+      const isRepoPath = isRepositoryPath(filePath);
+      fileButtons.enableButtons(isRepoPath);
 
       //Ajax request to check write permissions for file
-      $.ajax({
-        url: CONTEXT_PATH + 'api/repo/files/' + FileBrowser.encodePathComponents(filePath) + '/canAccessMap',
-        type: "GET",
-        beforeSend: function (request) {
-          request.setRequestHeader('accept', 'application/json');
-        },
-        data: {"permissions": "1|2"}, //check write and delete permissions for the given file
-        async: true,
-        cache: false,
-        success: function (response) {
-          fileButtons.updateFilePermissionButtons(response);
-        },
-        error: function (response) {
-          fileButtons.updateFilePermissionButtons(false);
-        }
-      });
+      if( filePath.charAt(0) == "/" ) {
+        filePath = Encoder.encodeRepositoryPath(filePath);
+
+        $.ajax({
+          url: CONTEXT_PATH + 'api/repo/files/' + FileBrowser.encodePathComponents(filePath) + '/canAccessMap',
+          type: "GET",
+          beforeSend: function (request) {
+            request.setRequestHeader('accept', 'application/json');
+          },
+          data: {"permissions": "1|2"}, //check write and delete permissions for the given file
+          async: true,
+          cache: false,
+          success: function (response) {
+            fileButtons.updateFilePermissionButtons(response);
+          },
+          error: function (response) {
+            fileButtons.updateFilePermissionButtons(false);
+          }
+        });
+      } else {
+        // this is a VFS connection path
+        //TODO BACKLOG-40086: refactor when we've implemented permissions for vfs connections folders/files
+      }
```

Comment:
You can leave the else if you want, but better to be commented out, imo.
Something like:
```js
// else
// This is a VFS connection path...
```